frontend/modules/sequences/sequences.selectors.ts

```diff
@@ -120,16 +120,26 @@ export const selectSelectedStartingDate = createSelector(
 );
 
 export const selectNextKeyFramesDates =  createSelector(
-	selectSelectedStartingDate, selectStepScale, selectStepInterval , selectMaxDate,
-		(startingDate, scale, interval, maxDate) => {
+	selectSelectedStartingDate, selectStepScale, selectStepInterval , selectMaxDate, selectFrames,
+		(startingDate, scale, interval, maxDate, frames) => {
 			const keyFrames = [];
 			keyFrames[0] = new Date(Math.min(maxDate, (startingDate || new Date(0)).valueOf()));
+			let lastFrame = getSelectedFrame(frames, keyFrames[0]);
+			let nextFrame = null;
 			let date = getDateByStep(startingDate, scale, interval);
 			maxDate = new Date(maxDate);
 
 			for (let i = 0; i < 3 ; i++) {
 				date = getDateByStep(date, scale, interval);
+				nextFrame = getSelectedFrame(frames, date);
```

Comment:
@sanmont3drepo  I think this can be simplified.

If I understood this correctly, we're just trying to find the next 3 key frames right, regardless of the step?

This can be achieved by just looking at the frames array in the sequence. Since we've moved out the activitiy entries, each frame entry in sequence.frames is guaranteed to be a unique frame state.

Reply:
The problem with doing that is the step, if you have a big step that spans through multiple frames, you would actually want just the steps that you are going through instead of fetching the inbetweens.